### PR TITLE
Encode signing output correctly

### PR DESCRIPTION
**Pull request by keefertaylor**
## Description

Fix encoding in signing output.
## Testing instructions

Run swift tests in XCode
## Types of changes
* Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ X ] Prefix PR title with `[WIP]` if necessary.
- [ X ] Add tests to cover changes as needed.
- [ X ] Update documentation as needed.

### CODE REVIEW

File: src/interface/TWTezosSigner.cpp [hunk 1]
```diff
@@ -51,7 +53,9 @@ TW_Tezos_Proto_SigningOutput TWTezosSignerSign(TW_Tezos_Proto_SigningInput data)
     auto signedBytesHex = signer.signOperationList(key, operationList);
```
Comment (alejandro-isaza) [inline src/interface/TWTezosSigner.cpp]: Can we generate this as a byte array to start with?

File: src/interface/TWTezosSigner.cpp [hunk 2]
```diff
@@ -51,7 +53,9 @@ TW_Tezos_Proto_SigningOutput TWTezosSignerSign(TW_Tezos_Proto_SigningInput data)
...
 
     auto protoOutput = Proto::SigningOutput();
-    protoOutput.set_signed_bytes(signedBytesHex);
+    TWString *signedBytesHexString = TWStringCreateWithUTF8Bytes(signedBytesHex.c_str());
+    TWData *signedBytes = TWDataCreateWithHexString(signedBytesHexString);
```
Comment (alejandro-isaza) [inline src/interface/TWTezosSigner.cpp]:
```suggestion
    auto signedBytes = parse_hex(signedBytesHex);
```

Comment (keefertaylor) [reply src/interface/TWTezosSigner.cpp]: Outdated.

File: src/interface/TWTezosSigner.cpp (hunk 1 above)
Comment (keefertaylor) [reply src/interface/TWTezosSigner.cpp]: Done.

**Comment by hewigovens** [comment]
I did go through Tezos `OperationList`, `Signer` and `Transaction` class.
It looks we convert from binary data (bytes) to hex string back and fourth a few times.
It should be more efficient to just return bytes in all `forge` functions

@keefertaylor and I think that's what @alejandro-isaza meant

File: src/Tezos/Address.cpp [hunk 3]
```diff
@@ -73,7 +75,9 @@ std::string Address::forge() const {
         int decodedLength = base58CheckDecodePrefix(s, prefixLength, prefix, decoded);
         if (decodedLength != 20)
             throw std::invalid_argument("Invalid Address For forge");
-        return "01" + TW::hex(decoded, decoded + decodedLength) + "00";
+        return parse_hex("01" + TW::hex(decoded, decoded + decodedLength) + "00");
```
Comment (alejandro-isaza) [inline src/Tezos/Address.cpp]:
```suggestion
        auto data = Data(); data.push_back(0x01); append(data, decoded); data.push_back(0);
```

File: src/Tezos/Address.cpp [hunk 4]
```diff
@@ -73,7 +75,9 @@ std::string Address::forge() const {
...
     }
-    return "00" + forgePublicKeyHash(s);
+    append(forged, parse_hex("00"));
```
Comment (alejandro-isaza) [inline src/Tezos/Address.cpp]:
```suggestion
    forged.push_back(0);
```

File: src/Tezos/Forging.cpp [hunk 5]
```diff
@@ -8,17 +8,20 @@
...
 
+#include <sstream>
+
 using namespace TW;
 
 // Forge the given boolean into a hex encoded string.
-std::string forgeBool(bool input) {
-    return input ? "ff" : "00";
+Data forgeBool(bool input) {
+    auto result = input ? "ff" : "00";
```
Comment (alejandro-isaza) [inline src/Tezos/Forging.cpp]:
```suggestion
    auto result = input ? 0xff : 0x00;
```

File: src/Tezos/Forging.cpp [hunk 6]
```diff
@@ -8,17 +8,20 @@
...
+    return parse_hex(result);
```
Comment (alejandro-isaza) [inline src/Tezos/Forging.cpp]:
```suggestion
    return Data{ result };
```

File: src/Tezos/Forging.cpp [hunk 7]
```diff
@@ -27,33 +30,58 @@ std::string forgePublicKeyHash(const std::string &publicKeyHash) {
     // Adjust prefix based on tz1, tz2 or tz3.
     switch ((char) publicKeyHash[2]) {
         case '1':
-            result += "0";
+            append(forged, parse_hex("0"));
```
Comment (alejandro-isaza) [inline src/Tezos/Forging.cpp]:
```suggestion
            forged.push_back(0)
```

File: src/Tezos/OperationList.cpp [hunk 8]
```diff
@@ -21,21 +21,21 @@ void OperationList::add_operation(Transaction transaction) {
...
-std::string OperationList::forgeBranch() const {
+Data OperationList::forgeBranch() const {
     size_t capacity = 128;
     uint8_t decoded[capacity];
     size_t prefixLength = 2;
     uint8_t prefix[] = {1, 52};
 
     int decodedLength = base58CheckDecodePrefix(branch, prefixLength, prefix, decoded);
-    return hex(decoded, decoded + decodedLength);
+    return parse_hex(hex(decoded, decoded + decodedLength));
```
Comment (alejandro-isaza) [inline src/Tezos/OperationList.cpp]:
```suggestion
    return decoded;
```

File: include/TrustWalletCore/TWStellarPassphrase.h [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+TW_EXTERN_C_BEGIN
+
+TW_EXPORT_ENUM()
+enum TWStellarPassphrase {
+    TWStellarPassphraseStellar     /* "Public Global Stellar Network ; September 2015" */,
+    TWStellarPassphraseKin         /* "Kin Mainnet ; December 2018" */,
+};
+
+static const char *_Nonnull TWStellarPassphrase_Stellar = "Public Global Stellar Network ; September 2015";
+static const char *_Nonnull TWStellarPassphrase_Kin = "Kin Mainnet ; December 2018";
```
Comment (alejandro-isaza) [inline include/TrustWalletCore/TWStellarPassphrase.h]: Let's use something more than an underscore to distinguish from enum cases. Maybe ` TWStellarPassphraseStellarString`

Comment (keefertaylor) [reply include/TrustWalletCore/TWStellarPassphrase.h]: This is just an auto-generated file that got merged in with the latest changes.

The file is this way at HEAD: https://github.com/TrustWallet/wallet-core/blob/d8d408c33085b9466ad007ed45f880ee43720d4e/include/TrustWalletCore/TWStellarPassphrase.h

**Comment by alejandro-isaza** [comment]
I'm seeing a lot of changes that are already on master, can you please rebase.

**Pull request closed without merging**